go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -175,7 +175,7 @@ func (ncc *networkClusterController) init() error {
 				ncc.NetInfo,
 				ncc.watchFactory.IPAMClaimsInformer().Lister(),
 			)
-			ncc.ipamClaimReconciler = ipamClaimsReconciler
+			ncc.ipamClaimReconciler = ipamClaimsReconciler.(*persistentips.IPAMClaimReconciler)
```

Comment:
Maybe it would have been cleaner:

```
ncc.ipamClaimReconciler = persistentips.NewIPAMClaimReconciler(
				ncc.kube,
				ncc.NetInfo,
				ncc.watchFactory.IPAMClaimsInformer().Lister(),
			)
ipamClaimsReconciler = ncc.ipamClaimReconciler
```

I am trying to recap how we ended up in this situation. I think we only wanted the interface with the `FindIPAMClaim` method exclusively to be used by the pod allocator. I lost track on why the `Reconcile` method also ended up in the interface and why we extended the use of the interface in the pod annotation allocator as well. I guess it was due to testing.

I think we should have taken the route of making the `allowsPersistentIPs` method more globally available, the we wouldn't have to use nil pointers as flags. For example, in `multinetwork.go`:
```
func DoesNetworkRequireTunnelIDs(netInfo NetInfo) bool {
	// Layer2Topology with IC require that we allocate tunnel IDs for each pod
	return netInfo.TopologyType() == types.Layer2Topology && config.OVNKubernetesFeature.EnableInterconnect
}
```

Reply:
I'll add a globally reachable helper we can re-use.

I'll push a separate follow-up PR with it.

Followed your suggestion.